src/main/java/clzzz/helper/logic/parser/ParserUtil.java

```diff
@@ -92,12 +91,9 @@ public static Species parseSpecies(String species) throws ParseException {
     public static DateOfBirth parseDateOfBirth(String dateOfBirth) throws ParseException {
         requireNonNull(dateOfBirth);
         String trimmedDateOfBirth = dateOfBirth.trim();
-        if (!DateOfBirth.isValidDateFormat(trimmedDateOfBirth)) {
+        if (!DateOfBirth.isValidDateOfBirth(trimmedDateOfBirth)) {
             throw new ParseException(DateOfBirth.MESSAGE_CONSTRAINTS);
         }
-        if (!DateOfBirth.isValidDate(trimmedDateOfBirth)) {
-            throw new ParseException(Messages.MESSAGE_INVALID_DATE);
-        }
```

Comment:
This deleted if statement is used for detecting invalid date. according to your changes now, the system will not be able to differentiate invalid format and invalid date...